packages/core/src/server.ts

```diff
@@ -0,0 +1,42 @@
+// Copyright IBM Corp. 2017. All Rights Reserved.
+// Node module: @loopback/core
+// This file is licensed under the MIT License.
+// License text available at https://opensource.org/licenses/MIT
+
+import { Context, Binding, Constructor } from '@loopback/context';
+import { Component, mountComponent } from './component';
+
+export class Server extends Context {
+  protected context: Context;
+  protected options?: ServerOptions;
+
+  // tslint:disable-next-line:no-any
+  constructor(app: Context, options?: ServerOptions, ...args: any[]) {
+    super(app);
+    this.options = options || {};
+
+    if (this.options.components) {
+      for (const component of this.options.components) {
+        this.component(component);
+      }
+    }
```

Comment:
As I wrote above, I believe components should be part of `Application`.

Do we want to have per-server components too? What use cases would require such feature?

Reply:
Maybe something that edits the Sequence for HTTP/S REST servers (but doesn't affect gRPC/etc)?

Reply:
In hindsight, it's not important that those components be server-specific, as long as they're not stomping about all over the place (but that would be the component developer's responsibility).

The parent-child propagation should ensure that each of the servers can retrieve whatever components are mounted at the application level, yes? @bajtos

Reply:
> The parent-child propagation should ensure that each of the servers can retrieve whatever components are mounted at the application level, yes?

Yes, that's correct.